### DISCUSSION TITLE
Issues running OrcaSlicer in Qubes OS

### DISCUSSION THREAD

**Discussion by ToddRAllen** [OrcaSlicer/OrcaSlicer#3881] [Q&A] [answered]
Qubes OS provides security through compartmentalization with Xen based virtualization.  All user applications such as OrcaSlicer are run in virtual machines called AppVMs.  I am running the latest stable Qubes, 4.2.0 and have tried both OrcaSlicer V1.9.0 and the latest daily builds and get the same behavior with respect to the following issues.  I have run OrcaSlicer in both of the bundled AppVMs based on Fedora 38 and Debian 12 as well as one I created based on a very stripped down Windows 10 Enterprise LTSC 2021.  The OrcaSlicer appimages ran on the stock Debian 12 AppVM but in Fedora 38 I had to install fuse and webkit2gtk4.0 before it would run.

In both linux based AppVMs it seems to function identically with the same issue.  In the Prepare screen I have been unable to use any of the 3 menus for Printer, Filament and Process.  Left mouse clicking on any of them appears to open the menu and instantly close it on the down stroke of the left mouse button.  I have looked and failed to find any alternative way to access these menus or their functions such as with keyboard shortcuts.  I am guessing virtualization is triggering an additional atypical event on the mouse down stroke which closes the menu.  This happens with all permutations I have tried such as running the app in a window or full screen and with various options for virtualization settings.  Any tips on steps I might take to further diagnose this issue would be appreciated.  I expect building OrcaSlicer from source and running in a debugger would clarify the problem but am not currently wanting to make that commitment of time and effort and I am looking easier things to try first.

This issue does not arise in the Windows AppVM running the OrcaSlicer portable builds where these menus all work as expected.  However I get a popup message stating a need for OpenGL 2.0 or later and there is no rendering of the build plate or access to the Preview screen and I don't know how to provide OpenGL support to my windows AppVM.  I am running on a Bee-Link MiniPC GTR7 7840HS.

**Comment by gmaxwell** [comment, marked as answer]
A less intrusive workaround is to run Xnest (with a window manager, like blackbox) in an appvm.  The dropdowns work.

**Reply by ToddRAllen** [reply]
Thanks!  I will check out Xnest.  I should add that I have been using OrcaSlicer under my Windows AppVM as I found that copying the file opengl32.dll that comes with PrusaSlicer for windows into my OrcaSlicer folder satisfied the requirement for OpenGL 2.0 or later.  I also got OrcaSlicer to work under linux AppVMs by disabling the virtualization integration tools although I found that too unpleasant to use.

**Reply by ToddRAllen** [reply]
Thanks!  I have been able to run OrcaSlicer under Xnest and can now use the various menus that were previously unusable.  This may become my primary slicing solution although I need to spend more time with it to see if I can refine it as the graphics interface is taking a bit of a hit.  I'm using Openbox as the window manager and there is quite a bit to experiment with between the various options of that, other window managers and options of Xnest.

Another option I have stumbled upon is that BambuStudio which suffered the same issue when run from a .appimage I have found works fine when run from a flatpak install.  I haven't used it much but it seems to work cleanly and if the differences with Orca aren't too painful I may just go with it.  Perhaps this is due to flatpak's sandboxing?  I wonder if Orca might also work properly if done as a flatpak?
